test/e2e/apimachinery/webhook.go

```diff
@@ -74,10 +83,29 @@ var _ = SIGDescribe("AdmissionWebhook", func() {
 		// Note that in 1.9 we will have backwards incompatible change to
 		// admission webhooks, so the image will be updated to 1.9 sometime in
 		// the development 1.9 cycle.
-		deployWebhookAndService(f, "gcr.io/kubernetes-e2e-test-images/k8s-sample-admission-webhook-amd64:1.8v2", context)
+		deployWebhookAndService(f, "gcr.io/kubernetes-e2e-test-images/k8s-sample-admission-webhook-amd64:1.8v3", context)
 		registerWebhook(f, context)
 		testWebhook(f)
 	})
+
+	It("Should be able to deny custom resource creation", func() {
+		// Make sure the relevant provider supports admission webhook
+		framework.SkipUnlessServerVersionGTE(serverWebhookVersion, f.ClientSet.Discovery())
+		framework.SkipUnlessProviderIs("gce", "gke", "local")
+		_, err := f.ClientSet.AdmissionregistrationV1alpha1().ValidatingWebhookConfigurations().List(metav1.ListOptions{})
+		if errors.IsNotFound(err) {
+			framework.Skipf("dynamic configuration of webhooks requires the alpha admissionregistration.k8s.io group to be enabled")
+		}
+		By("Setting up server cert")
+		namespaceName := f.Namespace.Name
+		context := setupServerCert(namespaceName, serviceName)
+		createAuthReaderRoleBinding(f, namespaceName)
+		deployWebhookAndService(f, "gcr.io/kubernetes-e2e-test-images/k8s-sample-admission-webhook-amd64:1.8v3", context)
+		crdCleanup, dynamicClient := createCRD(f)
+		framework.AddCleanupAction(crdCleanup)
+		registerWebhookForCRD(f, context)
```

Comment:
Does this mean we leak the webhook? Moreover, our CRD name is constant. What happens if you call the test again? We should at least make sure that this works. If there are flakes, running a test infinitly is usual practice. With the leak this does not work.

Reply:
I'm not sure I completely followed you. Are you asking if we leak the webhook configuration? It's removed in the [cleanWebhookTest](https://github.com/kubernetes/kubernetes/pull/55127/files#diff-fac90cb6856d5c08a931366f5169599cR403). The CRD is removed by `framework.AddCleanupAction(crdCleanup)`. 

I think `AddCleanupAction` is a wrong choice. CleanupAction is called after the entire suite completes. Instead, we should register all the cleanup functions in AfterEach. I'll fix that in another PR.

Reply:
Missed `cleanWebhookTest`. Then it's all fine.